stellargraph/connector/neo4j/graph.py

```diff
@@ -138,6 +139,48 @@ def to_adjacency_matrix(self, node_ids):
     def is_directed(self):
         return self._is_directed
 
+    def check_graph_for_ml(self, expensive_check=False):
+        """
+        Checks if all properties required for machine learning training/inference are set up.
+        An error will be raised if the graph is not correctly setup.
+        """
+
+        if expensive_check:
+            num_nodes_with_feats_query = f"""
+                MATCH(n)
+                WHERE "features" in keys(n)
+                return COUNT(*)
```

Comment:
This might get optimised to be equivalent, but this is probably more directly expressed as via https://neo4j.com/docs/cypher-manual/current/clauses/where/#property-existence-checking

```suggestion
                WHERE EXISTS(n.features)
                RETURN COUNT(*)
```

But it's probably even more efficient as:

```suggestion
                WHERE EXISTS(n.features)
                RETURN n LIMIT 1
```

and then checking `len(results.data())` or something. Since we only need to know that there's at least one node with features, rather than the exact count.

(Also, I've consistent-ified the capitalisation of the keywords.)